packages/schema/parse/src/validate/types.ts

```diff
@@ -149,3 +150,33 @@ export function propertyTypes(astNode: DocumentNode): void {
     }
   }
 }
+
+export function circularDefinitions(astNode: DocumentNode): void {
+  const operationTypes: string[] = [];
+  const operationTypeNames = ["Mutation", "Subscription", "Query"];
+
+  visit(astNode, {
+    enter: {
+      ObjectTypeDefinition: (node) => {
+        const isOperationType = operationTypeNames.some((name) =>
+          node.name.value.includes(name)
```

Comment:
This doesn't work for types like `type FooQueryOptions { ... }`. We instead want to look for `name === "Query" | "Mutation" | "Subscription"`, and also `name.endsWith("_Query" | "_Mutation" | "_Subscription")`

Reply:
Fixed in my updates